src/action.ts

```diff
@@ -329,6 +329,11 @@ class Action {
     core.debug('Generating artifact name')
     let artifactName = `assets-${process.env.GITHUB_JOB}`
 
+    const artifactNameExt = artifactName.split('').length 
+    if (artifactNameExt > 255) {
+      artifactName.slice(-10)
+    }
```

Comment:
Note that Javascript's [String](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/length) already provides a `length` property, so you don't need to create an array.

Also, when you [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) it, the original String doesn't mutates. Instead, it creates a new instance with the sliced value.

Therefore, you could try:

```suggestion
    if (artifactName.length > 255) {
      artifactName = artifactName.slice(-10)
    }
```

Reply:
Out of curious, why just slicing the latest 10 characters?